<?xml version="1.0"?>
<!--
  Software License Agreement (BSD)

  Copyright (c) 2019, WVU Interactive Robotics Laboratory
                        https://web.statler.wvu.edu/~irl/
  All rights reserved.

  Redistribution and use in source and binary forms, with or without
  modification, are permitted provided that the following conditions are met:

      * Redistributions of source code must retain the above copyright
        notice, this list of conditions and the following disclaimer.
      * Redistributions in binary form must reproduce the above copyright
        notice, this list of conditions and the following disclaimer in the
        documentation and/or other materials provided with the distribution.
      * Neither the name of the WVU Interactive Robotics Laboratory nor the names
        of its contributors may be used to endorse or promote products derived
        from this software without specific prior written permission.

  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  POSSIBILITY OF SUCH DAMAGE.
-->
<launch>

  <node pkg="twist_mux" type="twist_mux" name="twist_mux">
    <rosparam command="load" file="$(find pathfinder_control)/config/twist_mux.yaml" />
    <!-- <remap from="cmd_vel_out" to="cmd_vel"/> -->
  </node>

  <rosparam file="$(find pathfinder_control)/config/pathfinder_simple_sim_control.yaml" command="load" />

  <node name="controller_spawner" pkg="controller_manager" type="spawner" ns="pathfinder" args="joint_state_controller front_left_velocity_controller back_left_velocity_controller front_right_velocity_controller back_right_velocity_controller" />

  <node name="pathfinder_sim_control_node" pkg="pathfinder_control" type="pathfinder_sim_control_node" />

</launch>
